sdk/core/core-lro/test/utils/testOperation.ts

```diff
@@ -77,7 +78,7 @@ async function cancel(
 
   if (abortSignal && abortSignal.aborted) {
     // Simulating a try catch of an HTTP request that's given an aborted abortSignal.
-    return await this.update({
+    return this.update({
```

Comment:
This pattern feels really weird to me. We're getting the `no-invalid-this` error here because the use of `this` really is strange and impossible for the linter to tell that this is a method-like thing.

We could:
- refactor these "methods" and the result of `makeOperation` to be a class with true methods
- refactor them into functions that use a proper lexical binding of the operation instead of a `this` binding
- explicitly `bind` the value of `this` in the assignment of `create`, `update`, and `toString` in their assignment on L113-115

@xirzec thoughts?

Reply:
I agree that it feels weird. I'm not sure what the best solution is. 

If we want to use `this` I agree that we should likely make it a secret class, like how I get around exposing underlying classes in `core-https` by always returning the interface: https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/core/core-https/src/pipelineRequest.ts#L178

Otherwise, using `this` from a function feels bad, since it's making the function impure and harder to reason about. Typing `this` with TS would help a little bit, but maybe that's not common enough to feel natural?

Reply:
no-invalid-this causes issues with Key Vault and other clients too. Please let's not change this as part of this PR. We can log an issue to study approaches to fix no-invalid-this generally in our clients, or to remove this linting rule if we can't get through it.

Reply:
I created this issue https://github.com/Azure/azure-sdk-for-js/issues/11404.